lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller.pm

```diff
@@ -124,7 +37,7 @@ sub _check_error {
     return (400, "Subfolder has invalid characters") if $subfolder && $subfolder =~ m!/!;
     return (400, "Filename has invalid characters")  if $filename  && $filename =~ m!/!;
 
-    return (404, "Invalid Project") if $project && !-d Mojo::File->new($home, $project);
+    return (404, "Invalid Project {" . $project . "}") if $project && !-d Mojo::File->new($home, $project);
```

Comment:
With multiple controllers this method should probably be a helper.

Reply:
Why Controllers can't just derive it from common base class?

Reply:
Because using a helpers is the standard Mojolicious solution for sharing code between controllers. At least all code that doesn't fit into a model class, which some of your code might too. Controllers are very rarely subclassed (for various reasons, including security).

Reply:
I didn't notice this before, but `_check_error` is called as a function from `_check_and_render_error`, even though it's a method.

Reply:
Is it really big deal? It seems to work and I don't believe I have to create another helper to call private subs or move them away to separate file without much of reasons